pkg/health/checker/checker.go

```diff
@@ -437,27 +451,33 @@ func (h shadowTrafficHealthChecker) WatchService(
 						Service: serviceID,
 						Status:  resp.Health,
 					}
+					// send a copy because does otherwise there's a race condition where the healthResults change before it's read
+					// it doesn't matter for correctness, but the go test --race will complain
+					resultsCopy := healthResultsCopy(healthResults)
 					select {
 					case <-ctx.Done():
 						return
-					case resultCh <- healthResults:
+					case resultCh <- resultsCopy:
 					}
 				case nodeIDs = <-nodeIDsCh:
-					for _, nodeID := range nodeIDs {
```

Comment:
Was this the source of the bug? Sending `health.Passing` whenever a node id was received in `nodeIDsCh`?